src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs

```diff
@@ -43,7 +43,8 @@ public IDisposable Start(IServerInformation serverInformation, Func<IFeatureColl
             {
                 disposables.Add(engine.CreateServer(
                     address.Scheme,
-                    address.Host,
+                    // Unix sockets use a file path, not a hostname.
+                    address.Scheme == Constants.UnixScheme ? address.Path : address.Host,
```

Comment:
How do you specify the address since the host is ignored. Do you use triple forward slash? e.g:

    unix:///tmp/kestrel-test.sock/